base_interface.go

```diff
@@ -55,6 +61,38 @@ func requestNew(options options) rest.Request {
 	}
 }
 
+func ExtractEndpoint(link string) (string, error) {
+	parsedURL, err := url.Parse(link)
+	if err != nil {
+		return "", err
+	}
+
+	return parsedURL.Path, nil
+}
+
+func SetHost(request rest.Request, host string) (rest.Request, error) {
+	endpoint, err := ExtractEndpoint(request.BaseURL)
+	if err != nil {
+		return request, err
+	}
+
+	request.BaseURL = host + endpoint
+	return request, nil
+}
+
+func SetDataResidency(request rest.Request, region string) (rest.Request, error) {
+	regionalHost, isPresent := allowedRegionsHostMap[region]
+	if isPresent {
```

Comment:
it's generally better to wrap the `!ok` in the `if` block since it's a shorter path. You'll be able to avoid the nested `if` blocks and makes the code more readable. So something like 

```
regionalHost, ok := allowedRegionsHostMap[region]
if !ok {
  return error
}

do other stuff here
```

Reply:
Yeah that makes sense. Will correct it